### PR TITLE
Add numpy.histogram frontend

### LINKED ISSUE #16539: histogram

### PR DESCRIPTION
**Pull request by RohitSgh**
add_frontend_checklist
closes #16539

### CODE REVIEW

**Comment by RohitSgh** [comment]
Issue   
https://github.com/unifyai/ivy/issues/16539

File: ivy/functional/frontends/numpy/statistics/histograms.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@from_zero_dim_arrays_to_scalar
+def histogram(
+    a,
+    bins=10,
+    range=None,
+    normed=None,
+    weights=None,
+    density=None,
+    *,
+    out=None,
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/statistics/histograms.py:22]: not sure if I overlooked anything, but I don't think the numpy histogram has out and dtype argument?
ref: https://numpy.org/doc/stable/reference/generated/numpy.histogram.html

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey there! Thanks for the great work so far! I've left a small comment below, feel free to ping me if there's any issue or question! Thanks!

**Comment by RohitSgh** [comment]
add_frontend_checklist

**Comment by RohitSgh** [comment]
Can you please help me in resolving the issue, @juliagsy? At least 5 checks have failed so far.

**Comment by RohitSgh** [comment]
Tried to solve on own and things getting worse 😅   
Please help @juliagsy in your free time

**Comment by RohitSgh** [comment]
add_frontend_checklist

**Comment by juliagsy** [comment]
Hey @RohitSgh ! So sorry, seems like there might be some issue with the ci earlier, would you mind pushing a commit of some dummy changes so that the ci reruns automatically? Apologies for any inconvenience and thanks a lot!

File: ivy/functional/frontends/numpy/statistics/histograms.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@from_zero_dim_arrays_to_scalar
+def histogram(
+    a,
+    *,
+    bins=10,
+    range=None,
+    density=None,
+    weights=None,
+):
+    [x.dtype for x in [a, weights]]
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/statistics/histograms.py:18]: as the ivy histogram function are able to handle the default values from numpy, could you please try to reduce additional codes here?
unless there are necessary for the tests to pass, please try to minimise additional code!

File: ivy/functional/frontends/numpy/statistics/histograms.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    if weights is None:
+        weights = ivy.ones_like(a)
+    if range is None:
+        range = (a.min(), a.max())
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/statistics/histograms.py:22]: one necessary condition might be if `density` is `None`, we set it to `False`

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey! Thanks for contributing! I've left some comments below, please also make sure that all tests are passing after that! Thanks!

**Comment by RohitSgh** [comment]
Hi @juliagsy. I will work on some different issue. Therefore, Closing PR. Thanks!

**Pull request closed without merging**